### PR TITLE
Feature: selection on click

### PR DESCRIPTION
**Pull request by Zorin95670**
:information_source:  Don't forget to modify the changelog.md before merging this branch.

### CODE REVIEW

File: src/draw/DefaultDrawer.js [hunk 1]
```diff
@@ -25,6 +25,19 @@ class DefaultDrawer {
      * @type {Number}
      */
     this.margin = 10;
+    this.actions = {
```
Comment (Zorin95670) [inline src/draw/DefaultDrawer.js:28]: missing documentation

File: src/draw/DefaultDrawer.js [hunk 2]
```diff
@@ -25,6 +25,19 @@ class DefaultDrawer {
...
+        style: '2px solid hsl(205, 100%, 50%)',
+        offset: '3px',
+      },
+      drag: {
+        state: false,
+        deltaX: 0,
+        deltay: 0,
+      },
+    };
+    this.emit = () => {};
```
Comment (Zorin95670) [inline src/draw/DefaultDrawer.js:40]: remove this

File: src/draw/DefaultDrawer.js [hunk 3]
```diff
@@ -67,13 +80,94 @@ class DefaultDrawer {
...
+    components.forEach((component) => {
+      this.setComponentAction(component);
+    });
 
     componentsSelection.exit()
       .remove();
     return true;
   }
 
+  setViewPortAction(element) {
```
Comment (Zorin95670) [inline src/draw/DefaultDrawer.js:94]: missing documentation

File: src/draw/DefaultDrawer.js [hunk 4]
```diff
@@ -67,13 +80,94 @@ class DefaultDrawer {
...
+    element.on('mousedown', () => {
+      this.__unselectComponent();
+    });
+  }
+
+  setComponentAction(component) {
```
Comment (Zorin95670) [inline src/draw/DefaultDrawer.js:100]: missing documentation

File: src/draw/DefaultDrawer.js [hunk 5]
```diff
@@ -67,13 +80,94 @@ class DefaultDrawer {
...
+        .on('end', (event) => {
+          if (!this.actions.drag.state) {
+            this.__toggleComponentSelection(event.subject.id);
+          }
+          this.__dragEnd(event, component, element);
+        }),
+    );
+  }
+
+  __dragStart(event, component, element) {
```
Comment (Zorin95670) [inline src/draw/DefaultDrawer.js:120]: missing documentation

File: src/draw/DefaultDrawer.js [hunk 6]
```diff
@@ -67,13 +80,94 @@ class DefaultDrawer {
...
+    this.actions.drag.state = false;
+    this.d3.select('use')
+      .attr('xlink:href', `#${component.id}`);
+    this.actions.drag.deltaX = event.x - element.attr('x');
+    this.actions.drag.deltaY = event.y - element.attr('y');
+    element.attr('cursor', 'grabbing');
+  }
+
+  __dragPending(event, component, element) {
```
Comment (Zorin95670) [inline src/draw/DefaultDrawer.js:129]: missing documentation

File: src/draw/DefaultDrawer.js [hunk 7]
```diff
@@ -67,13 +80,94 @@ class DefaultDrawer {
...
+    this.actions.drag.state = true;
+    const x = event.x - this.actions.drag.deltaX;
+    const y = event.y - this.actions.drag.deltaY;
+
+    element.attr('x', x).attr('y', y);
+    component.drawOption.x = x;
+    component.drawOption.y = y;
+  }
+
+  __dragEnd(event, component, element) {
```
Comment (Zorin95670) [inline src/draw/DefaultDrawer.js:139]: missing documentation

File: src/draw/DefaultDrawer.js [hunk 8]
```diff
@@ -67,13 +80,94 @@ class DefaultDrawer {
...
+    element.attr('cursor', 'grab');
+    this.actions.drag.state = false;
+    this.actions.drag.deltaX = 0;
+    this.actions.drag.deltaY = 0;
+  }
+
+  __unselectComponent() {
```
Comment (Zorin95670) [inline src/draw/DefaultDrawer.js:146]: missing documentation

File: src/draw/DefaultDrawer.js [hunk 9]
```diff
@@ -67,13 +80,94 @@ class DefaultDrawer {
...
+    if (this.actions.selection.current) {
+      this.d3.select(`#${this.actions.selection.current}`)
+        .style('outline', '');
+      this.actions.selection.current = null;
+    }
+  }
+
+  __selectComponent(id) {
```
Comment (Zorin95670) [inline src/draw/DefaultDrawer.js:154]: missing documentation

File: src/draw/DefaultDrawer.js [hunk 10]
```diff
@@ -67,13 +80,94 @@ class DefaultDrawer {
...
+    this.__unselectComponent();
+
+    this.d3.select(`#${id}`)
+      .style('outline', this.actions.selection.style)
+      .style('outline-offset', this.actions.selection.offset);
+    this.actions.selection.current = id;
+  }
+
+  __toggleComponentSelection(id) {
```
Comment (Zorin95670) [inline src/draw/DefaultDrawer.js:163]: missing documentation

File: src/draw/DefaultDrawer.js [hunk 11]
```diff
@@ -37,7 +53,7 @@ class DefaultDrawer {
     const componentsSelection = this.d3.select(parentId)
       .selectAll('.component')
       .data(components, (data) => data);
-    const svgContainer = componentsSelection.enter().append('svg');
+    const svgContainer = componentsSelection.enter().append('svg').style('border-radius', '1px');
```
Comment (Hundraw) [inline src/draw/DefaultDrawer.js:56]: I think it's not really "pretty" to define the border-radius here without having the context.

**Review by Hundraw — CHANGES_REQUESTED** [review]

**Review by Hundraw — APPROVED** [review]

**Review by Hundraw — APPROVED** [review]

**Pull request merged**